pyfakefs/fake_filesystem.py

```diff
@@ -2915,6 +2924,21 @@ def _add_standard_streams(self) -> None:
         self._add_open_file(StandardStreamWrapper(sys.stdout))
         self._add_open_file(StandardStreamWrapper(sys.stderr))
 
+    def _create_temp_dir(self):
+        if not self.create_temp_dir:
```

Comment:
I would guard `_create_temp_dir` from the `reset` method instead, especially if `reset` can be called from `__init__`.

Reply:
Yes - I decided to use this one, because otherwise I would have to guard it twice. But if I call `reset` from `__init__` instead, this won't be true anymore.